### PR TITLE
pacific: Backport of bucket notifications tests

### PR DESCRIPTION
**Pull request by TRYTOBE8TME**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by yuvalif** [comment]
can you please paste a link to passing teuthology run?

**Comment by yuvalif** [comment]
in the commits where you had conflicts, please remove the following text fro the commit comment (keep all the rest of your comments):
```
It looks like you may be committing a cherry-pick.
 If this is not correct, please run
	git update-ref -d CHERRY_PICK_HEAD
 and try again.
```

File: qa/suites/rgw/notifications/tasks/test_amqp.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+tasks:
+- rabbitmq:
+    client.0:
+- notification-tests:
+    client.0:
+      force-branch: wip-backport-to-pacific
+      git_remote: https://github.com/TRYTOBE8TME/
```
Comment (yuvalif) [inline qa/suites/rgw/notifications/tasks/test_amqp.yaml:7]: don't forget to fix that before merge

File: qa/suites/rgw/notifications/tasks/test_kafka.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+tasks:
+- kafka:
+    client.0:
+      kafka_version: 2.6.0
+- notification-tests:
+    client.0:
+      force-branch: wip-backport-to-pacific
+      git_remote: https://github.com/TRYTOBE8TME/
```
Comment (yuvalif) [inline qa/suites/rgw/notifications/tasks/test_kafka.yaml:8]: don't forget to fix that before merge

File: qa/suites/rgw/notifications/tasks/test_others.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+tasks:
+- notification-tests:
+    client.0:
+      force-branch: wip-backport-to-pacific
+      git_remote: https://github.com/TRYTOBE8TME/
```
Comment (yuvalif) [inline qa/suites/rgw/notifications/tasks/test_others.yaml:5]: don't forget to fix that before merge

File: qa/suites/rgw/notifications/tasks/test_amqp.yaml (hunk 1 above)
Comment (TRYTOBE8TME) [reply qa/suites/rgw/notifications/tasks/test_amqp.yaml:7]: Yup, I'll take care.

**Comment by TRYTOBE8TME** [comment]
> can you please paste a link to passing teuthology run?

Teuthology link: https://pulpito.ceph.com/kapandya-2021-06-14_15:42:05-rgw:notifications-wip-bntests-backport-distro-basic-smithi/

**Review by yuvalif — APPROVED** [review]

**Comment by cbodley** [comment]
please rebase

**Comment by yuriw** [comment]
@TRYTOBE8TME pls rebase and add `needs-qa`, thx

**Comment by yuvalif** [comment]
> @TRYTOBE8TME can you please rebase this, and pull in your extra fix from #43995?

@yuriw could you please sync up with @TRYTOBE8TME to see when this could enter the test queue.
so, we avoid constant rebasing of this PR.

**Comment by yuriw** [comment]
@TRYTOBE8TME pls rebase

**Comment by cbodley** [comment]
failed qa in http://qa-proxy.ceph.com/teuthology/yuriw-2022-08-03_17:55:17-rgw-wip-yuri3-testing-2022-08-03-0828-pacific-distro-default-smithi/6957156/teuthology.log
```
2022-08-03T21:33:23.021 INFO:teuthology.orchestra.run.smithi036.stderr:test pushing amqp s3 notification on master ... FAIL
2022-08-03T21:33:30.792 INFO:teuthology.orchestra.run.smithi036.stderr:bucket_notification.test_bn.test_ps_s3_creation_triggers_on_master ... FAIL
2022-08-03T21:33:36.937 INFO:teuthology.orchestra.run.smithi036.stderr:test multipart object upload on master ... FAIL
2022-08-03T21:33:42.946 INFO:teuthology.orchestra.run.smithi036.stderr:test s3 notification of metadata on master ... FAIL
2022-08-03T21:33:53.907 INFO:teuthology.orchestra.run.smithi036.stderr:test s3 notification of tags on master ... ok
2022-08-03T21:33:59.423 INFO:teuthology.orchestra.run.smithi036.stderr:test s3 notification of object versions ... ok
2022-08-03T21:34:07.133 INFO:teuthology.orchestra.run.smithi036.stderr:test s3 notification of deletion markers on master ... FAIL
2022-08-03T21:34:49.428 INFO:teuthology.orchestra.run.smithi036.stderr:test pushing persistent notification amqp ... FAIL
2022-08-03T21:35:31.770 INFO:teuthology.orchestra.run.smithi036.stderr:test pushing persistent notification of large notifications ... FAIL
```

**Review by cbodley — DISMISSED** [review]
failed qa

**Comment by yuriw** [comment]
@TRYTOBE8TME pls add `needs-qa` tag when it's ready for testing

**Comment by kamoltat** [comment]
Rados Approved!

[Pulpito :: Results Dashboard](https://pulpito.ceph.com/yuriw-2022-08-04_11:58:29-rados-wip-yuri3-testing-2022-08-03-0828-pacific-distro-default-smithi/)

0 Related Failures:
17 Tracked Failures:

2 jobs with [Bug #53501: Exception when running 'rook' task. - Orchestrator - Ceph](https://tracker.ceph.com/issues/53501)
1 job with [Bug #53939: ceph-nfs-upgrade, pacific: Upgrade Paused due to UPGRADE_REDEPLOY_DAEMON: Upgrading daemon osd.0 on host smithi103 failed - Orchestrator - Ceph](https://tracker.ceph.com/issues/53939)
3 jobs with [Bug #55443: "SELinux denials found.." in rados run - Infrastructure - Ceph](https://tracker.ceph.com/issues/55443)
1 job with [Bug #45318: Health check failed: 2/6 mons down, quorum b,a,c,e (MON_DOWN)" in cluster log running tasks/mon_clock_no_skews.yaml - RADOS - Ceph](https://tracker.ceph.com/issues/45318)
2 jobs with [Bug #52321: qa/tasks/rook times out: 'check osd count' reached maximum tries (90) after waiting for 900 seconds - Orchestrator - Ceph](https://tracker.ceph.com/issues/52321)
1 job with [Bug #54071: rados/cephadm/osds: Invalid command: missing required parameter hostname(<string>) - Orchestrator - Ceph](https://tracker.ceph.com/issues/54071)
1 job with [Bug #52124: Invalid read of size 8 in handle_recovery_delete() - RADOS - Ceph](https://tracker.ceph.com/issues/52124)
1 job with [Bug #56573: test_cephadm.sh: KeyError: 'TYPE' - Orchestrator - Ceph](https://tracker.ceph.com/issues/56573)

0 Newly Tracked Failures:
0 Dead Jobs

File: src/test/rgw/bucket_notification/test_bn.py [hunk 4]
```diff
@@ -0,0 +1,3087 @@
...
+            '''))
+        os.environ['RABBITMQ_CONFIG_FILE'] = os.path.splitext(RABBITMQ_CONF_FILE)[0]
+
+        ps_s3_creation_triggers_on_master(ca_location=CACERTFILE)
+
+        del os.environ['RABBITMQ_CONFIG_FILE']
+
+
+@attr('amqp_test')
+def test_ps_s3_multipart_on_master():
```
Comment (yuvalif) [inline src/test/rgw/bucket_notification/test_bn.py:1594]: please change according to: https://github.com/ceph/ceph/blob/main/src/test/rgw/bucket_notification/test_bn.py#L2069

the multipart fix was backported. see: 64bb3274756f0caa491c6cb34cdeca1b981baf3b

File: src/test/rgw/bucket_notification/test_bn.py [hunk 5]
```diff
@@ -0,0 +1,3087 @@
...
+    topic_conf2.del_config()
+    topic_conf3.del_config()
+    for key in bucket.list():
+        key.delete()
+    # delete the bucket
+    conn.delete_bucket(bucket_name)
+
+
+@attr('amqp_test')
+def test_ps_s3_metadata_on_master():
```
Comment (yuvalif) [inline src/test/rgw/bucket_notification/test_bn.py:1685]: there is a test issue that was fixed in: https://github.com/ceph/ceph/blob/main/src/test/rgw/bucket_notification/test_bn.py#L2158

the check in lines 1751 was replaced with:
https://github.com/ceph/ceph/blob/main/src/test/rgw/bucket_notification/test_bn.py#L2239

File: src/test/rgw/bucket_notification/test_bn.py [hunk 6]
```diff
@@ -0,0 +1,3087 @@
...
+    s3_notification_conf.del_config()
+    topic_conf.del_config()
+    # delete the bucket
+    bucket.delete_key(key.name, version_id=ver2)
+    bucket.delete_key(key.name, version_id=ver1)
+    conn.delete_bucket(bucket_name)
+
+
+@attr('amqp_test')
+def test_ps_s3_versioned_deletion_on_master():
```
Comment (yuvalif) [inline src/test/rgw/bucket_notification/test_bn.py:1931]: test issue that was fixed in:
https://github.com/ceph/ceph/blob/main/src/test/rgw/bucket_notification/test_bn.py#L2445

**Review by anthonyeleven — APPROVED** [review]

**Comment by dang** [comment]
@cbodley Can we merge this?

**Comment by cbodley** [comment]
@TRYTOBE8TME it looks like there are unresolved comments about `git_remote: https://github.com/TRYTOBE8TME/`

**Comment by TRYTOBE8TME** [comment]
jenkins test make check

**Comment by TRYTOBE8TME** [comment]
jenkins test make check

**Comment by yuvalif** [comment]
@cbodley do we still need to backport tests to pacific?

**Comment by cbodley** [comment]
> @cbodley do we still need to backport tests to pacific?

@yuvalif we'll likely do one or two more point releases for pacific. but these bucket notification tests may not be worth the effort

**Comment by yuvalif** [comment]
closing, according to the comment above

**Pull request closed without merging**